### PR TITLE
Improve playground and test

### PR DESCRIPTION
**Pull request by july2993**
### What problem does this PR solve? 
- Fix the issue that playground display can not show correct uptime status if the process is exited.
- Fix the issue that playground keep silent if a process is exited with non zero exit code, It should Tail the latest log after #515
- better make sure kill all process and improve test.

this is broken after #588 

### What is changed and how it works?
Do not call the Cmd.Process.Wait directly or we can not get the
Cmd.ProcessState and the Uptime show in display can't find out
the process is quit. Also, Cmd.Process return the ProcessState and
nil if the process exit with not 0 code, so if the instance quits
unexpectedly will tail the last log of the instance will never work.

Setup signal handler before we boot the cluster. Before this PR, in bootCluster, if
return before we setup the signal handler and receive signal(like user
press ctrl-c), playground will quit directly, but we may have started
some subprocess which become orphaned process.

Improve the integration test to make sure scale-in/out, display
basically work.

### Check List 

Tests 

 - Integration test
 - Manual test (add detailed scripts or steps below)

Release notes:

```release-note
NONE
```

### CODE REVIEW

File: components/playground/grafana.go [hunk 1]
```diff
@@ -149,6 +152,7 @@ func makeSureDir(fname string) error {
 var clusterName string = "playground"
 
 // dir should contains files untar the grafana.
+// return not error iff the Cmd is started successfully.
```
Comment (lonng) [inline components/playground/grafana.go:155]:
```suggestion
// return not error if the Cmd is started successfully.
```

Comment (july2993) [reply components/playground/grafana.go:155]: iff = if and only if

File: components/playground/instance/process.go [hunk 2]
```diff
@@ -50,8 +57,9 @@ func (p *process) Pid() int {
 // Uptime implements Instance interface.
 func (p *process) Uptime() string {
 	s := p.cmd.ProcessState
-	if s != nil && s.Exited() {
-		return "exited"
+
+	if s != nil {
```
Comment (lonng) [inline components/playground/instance/process.go:61]: Can we retrieve the `ProcessState` if the current process is running? If so this method won't return the real `Uptime`.

File: components/playground/main.go [hunk 3]
```diff
@@ -126,7 +130,66 @@ Examples:
...
+				go p.terminate(sig)
+				// If user try double ctrl+c, force quit
+				sig = (<-sc).(syscall.Signal)
+				atomic.StoreInt32(&p.curSig, int32(syscall.SIGKILL))
+				if sig == syscall.SIGINT {
+					p.terminate(syscall.SIGKILL)
+				}
+			}()
+
+			// TODO: we can set Pdeathsig of Cmd.SysProcAttr(linux only) in all the Cmd we started to kill
```
Comment (lonng) [inline components/playground/main.go:171]: Can we implement this in the current PR?

File: components/playground/playground.go [hunk 4]
```diff
@@ -46,27 +44,33 @@ import (
...
+	tidbs            []*instance.TiDBInstance
+	tiflashs         []*instance.TiFlashInstance
+	pumps            []*instance.Pump
+	drainers         []*instance.Drainer
+	startedInstances []instance.Instance
 
 	idAlloc        map[string]int
 	instanceWaiter errgroup.Group
 
+	// not nil iff we start the exec.Cmd successfully.
```
Comment (lonng) [inline components/playground/playground.go:69]:
```suggestion
	// not nil if we start the exec.Cmd successfully.
```

File: components/playground/instance/process.go (hunk 2 above)
Comment (july2993) [reply components/playground/instance/process.go:61]: 131     // ProcessState contains information about an exited process,
132     // available after a call to Wait or Run.
133     ProcessState *os.ProcessState

It is only available after the proccss exited and we call Wait on the Cmd.

File: components/playground/main.go (hunk 3 above)
Comment (july2993) [reply components/playground/main.go:171]: #719 #720 
I will try but not in this PR.
Keeping this pr small and do one thing.

**Review by lonng — APPROVED** [review]
LGTM
